lib/tools/phpunit/CodeAuditTestCase.class

```diff
@@ -0,0 +1,511 @@
+<?php
+
+/*
+ * Gallery - a web based photo album viewer and editor
+ * Copyright (C) 2000-2008 Bharat Mediratta
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation;
+ * either version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street - Fifth Floor,
+ * Boston, MA  02110-1301, USA.
+ */
+
+/**
+ * A framework for scanning all of our source files for specific patterns.  Subclasses must override
+ * the checkFile() method to implement their own checks.
+ * @package Gallery
+ * @subpackage PHPUnit
+ * @author Bharat Mediratta <bharat@menalto.com>
+ * @version $Revision: 17580 $
+ */
+class CodeAuditTestCase extends GalleryTestCase {
+	public function __construct($methodName) {
+		parent::__construct($methodName);
+
+		$this->_shouldHalt = false;
+		$this->_baseDir    = realpath(__DIR__ . '/../../..');
+		$this->_checkTime  = time();
+	}
+
+	public function testCodeAudit() {
+		global $gallery;
+
+		$platform =& $gallery->getPlatform();
+
+		if (!$platform->file_exists($this->_baseDir . '/.svn/entries')) {
+			//$this->skip();
+			return;
+		}
+
+		if (!function_exists('xml_parser_create')) {
+			$this->skip();
+
+			return;
+		}
+
+		$this->_scanDir($this->_baseDir);
+
+		if ($this->_shouldHalt) {
+			$this->skip();
+
+			return;
+		}
+
+		$this->finishScan($this->_shouldHalt);
+	}
+
+	public function _scanDir($dirPath) {
+		global $gallery;
+
+		$platform =& $gallery->getPlatform();
+
+		if (($handle = $platform->fopen($dirPath . '/.svn/entries', 'r')) === false) {
+			$this->skip();
```

Comment:
To be honest, I don't know the purpose of this code.

It looks like it is there to check if there is a SVN tree. As we are not using SVN any more, it might be we should "translate" it to a GIT path, but not quite sure about the purpose.

Anyway, I will comment it too.